### PR TITLE
release memory

### PR DESCRIPTION
**Pull request by wyklion**
I dont think this smartArray is suit for JS...

### CODE REVIEW

File: src/Tools/babylon.smartArray.ts [hunk 1]
```diff
@@ -44,6 +44,7 @@
         }
 
         public reset(): void {
+            this.data.length = 0;
```
Comment (deltakosh) [inline src/Tools/babylon.smartArray.ts]: the whole point of the smartArray is to AVOID this ;)

**Comment by deltakosh** [comment]
WHy this?

File: src/Tools/babylon.smartArray.ts (hunk 1 above)
Comment (nockawa) [inline src/Tools/babylon.smartArray.ts]: @deltakosh check slack's #general

**Comment by deltakosh** [comment]
I did this to address memory: https://github.com/BabylonJS/Babylon.js/commit/f59cb9e1236cd750c1bdba4cfbac4a17d7acce66

**Comment by wyklion** [comment]
yes. dispose is better.

**Pull request closed without merging**